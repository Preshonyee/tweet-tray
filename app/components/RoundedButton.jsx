import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'styled-components';

import * as constants from '../constants';

const RoundedButtonStyle = Styled.button`
  user-select: none;
  display: block;
  margin: 0 auto;
  height: 38px;
  padding-left: 24px;
  padding-right: 24px;
  border-radius: 100px;
  font-size: ${constants.REGULAR_FONT_SIZE}px;
  font-weight: bold;
  border: 0px;
  outline: 0px;
  background-color: ${(props) => {
    return props.color;
  }};
  color: ${constants.WHITE};
  transition: all 0.3s ease 0s;
  line-height: 20px;
  text-align: center;

  &.fullWidth {
      width: 100% !important;
  }

  &.borderButton {
      color: ${constants.BLUE} !important;
      border: 2px solid ${constants.BLUE} !important;
      background-color: ${constants.WHITE} !important;
  }

  &:hover {
      opacity: 0.75;
      cursor: pointer;
      transition: all 0.3s ease 0s;
  }

  &:disabled, &[disabled] {
    opacity: 0.4 !important;
    cursor: default !important;
}
`;

const RoundedButton = (props) => {
  const {
    disabled,
    theme,
    color,
    marginTop,
    height,
    title,
    onClick,
    borderButton,
    fullWidth,
    type,
  } = props;

  return (
    <RoundedButtonStyle
      theme={theme}
      color={color}
      type={type}
      style={{
        marginTop: `${marginTop}px`,
        height: `${height}px`,
      }}
      className={`${fullWidth ? 'fullWidth' : ''} ${borderButton ? 'borderButton' : ''}`}
      onClick={onClick}
      disabled={disabled}
      value={title}
    >
      {title}
    </RoundedButtonStyle>
  );
};

RoundedButton.propTypes = {
  disabled: PropTypes.bool,
  theme: PropTypes.string,
  color: PropTypes.string,
  borderButton: PropTypes.bool,
  fullWidth: PropTypes.bool,
  marginTop: PropTypes.number,
  height: PropTypes.number,
  title: PropTypes.string,
  onClick: PropTypes.func,
  type: PropTypes.string,
};

RoundedButton.defaultProps = {
  disabled: false,
  theme: 'day',
  color: constants.BLUE,
  borderButton: false,
  fullWidth: false,
  marginTop: 0,
  height: 38,
  title: 'A Button',
  onClick: {},
  type: 'button',
};

export default RoundedButton;

