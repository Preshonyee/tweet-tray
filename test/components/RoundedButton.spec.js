import React from 'react';
import Enzyme, { shallow, } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RoundedButton from '../../app/components/RoundedButton';

Enzyme.configure({ adapter: new Adapter(), });

function setup() {
  const component = shallow(<RoundedButton />);
  return {
    component,
  };
}

describe('RoundedButton component', () => {
  it('should display button with title', () => {
    const { component, } = setup();
    expect(component.prop('value')).toMatchSnapshot();
  });
});
