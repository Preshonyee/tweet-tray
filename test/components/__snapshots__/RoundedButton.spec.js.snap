// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedButton component button should contain value 1`] = `"A Button"`;

exports[`RoundedButton component should render 1`] = `
<RoundedButton__RoundedButtonStyle
  className="fullWidth "
  color="#1da1f2"
  disabled={false}
  onClick={[MockFunction]}
  style={
    Object {
      "height": "38px",
      "marginTop": "0px",
    }
  }
  theme="day"
  type="button"
  value="A Button"
>
  A Button
</RoundedButton__RoundedButtonStyle>
`;

exports[`RoundedButton component should render full width button 1`] = `
<RoundedButton__RoundedButtonStyle
  className="fullWidth "
  color="#1da1f2"
  disabled={false}
  onClick={[MockFunction]}
  style={
    Object {
      "height": "38px",
      "marginTop": "0px",
    }
  }
  theme="day"
  type="button"
  value="A Button"
>
  A Button
</RoundedButton__RoundedButtonStyle>
`;
